### PR TITLE
f-analytics@0.10.0 - Moved push 'PageData' from mixin to service

### PR DESCRIPTION
**Pull request by oliversweb**
### Changed
- Moved the prepPageData from Mixin into Service to allow the consumer to dictate when to push the GA data.
- Remove redundant Mixin
- Updated readme

### CODE REVIEW

File: packages/services/f-analytics/README.md [hunk 1]
```diff
@@ -16,106 +16,56 @@ Encapsulates the GTM / Google Analytics functionality
...
 
 ---
-This component abstracts away the gathering of the various data values needed for Google Analytics (GA) and the setting up of Google Tag Manager (GTM).  Once registered it will; on render, push these various data values to GA.  You can see the results by inspecting the `dataLayer` from the browser console in developer tools and looking for the model `{platformData: {…}}`.
+This component abstracts away the gathering of the various data values needed for Google Analytics (GA) and the setting up of Google Tag Manager (GTM).<br>
+Once registered and the site is running it will prepare each page with the required GTM tags for pushing analytics to GA plus it will gather and store any serverside values to be used in analytics that are pushed to GA clientside.<br>
+You can see the GTM tags and any GA data by inspecting the `header` of the page and the `dataLayer` from the browser console in developer tools.  If you excerise all the current functionality you will be able to see the following models `{platformData: {…}}`, `{userData: {…}}`, `{pageData: {…}}` & `{event: ...}` in the `dataLayer`.
```
Comment (dandel10n) [inline packages/services/f-analytics/README.md]:
```suggestion
You can see the GTM tags and any GA data by inspecting the `header` of the page and the `dataLayer` from the browser console in developer tools.  If you exercise all the current functionality you will be able to see the following models `{platformData: {…}}`, `{userData: {…}}`, `{pageData: {…}}` & `{event: ...}` in the `dataLayer`.
```

File: packages/services/f-analytics/README.md [hunk 2]
```diff
@@ -16,106 +16,56 @@ Encapsulates the GTM / Google Analytics functionality
...
 
 
-## Benefits (Now)
-- Single point to record GA data: Currently we have GA/GTM logic scattered throught various features, this will allow that logic to be removed and centralise it a single component.
-- Self-sufficient: With only supplying a small amount of global data this component will attempt to evaluate, gather and record all the data required for the GA `platformData`, `userData` and `pageData` models.
-- Provide the facility to push 'ad-hoc' GA events via a global service (even if serverside).
+## Benefits
+- Single component to record GA data rather than having logic & implementations scattered around in various features.
+- Self-sufficient: With only supplying a small amount of data this component will attempt to evaluate, gather and record all the data required (serverside and clientside) for the GA models; `platformData`, `userData` and `pageData`.
```
Comment (dandel10n) [inline packages/services/f-analytics/README.md]:
```suggestion
- Self-sufficient: With only supplying a small amount of data this component will attempt to evaluate, gather and record all the data required (serverside and clientside) for the GA models: `platformData`, `userData` and `pageData`.
```

File: packages/services/f-analytics/README.md [hunk 3]
```diff
@@ -16,106 +16,56 @@ Encapsulates the GTM / Google Analytics functionality
...
-    As this mixin uses authentication token for some of its data it is needed for it to be added on the page level. There are three public methods available to use:
-    - `preparePlatformData` - evaluates and gather data for the `platformData` GA model.
-    - `prepareUserData` - evaluates and gather data for the `userData` GA model. Can take optional `authToken` prop to push data related to authToken. Without the prop userData will contain only userId.
-    - `preparePageData` - evaluates and gather data for the `pageData` GA model. Can take optional object containing `conversationId, requestId, authToken` prop to push data related to these attributes.
-    - `pushAnalytics` - pushes `platformData` and `userData` to the dataLayer.
+    F-Analytics is a 'Nuxt - Plugin' that needs to be registered in the "nuxt.config.js" but as it is contained in a fozzie service you need to create a local plugin to reference the external plugin, which then allows you to pass in options (see the `Options` section).<br><br>
+    In the example below it demonstrates how to create a local plugin; that references the `f-analytics` plugin, and how you can create the `options` to pass into the plugin;
```
Comment (dandel10n) [inline packages/services/f-analytics/README.md]:
```suggestion
    In the example below it demonstrates how to create a local plugin that references the `f-analytics` plugin, and how you can create the `options` to pass into the plugin;
```

File: packages/services/f-analytics/README.md [hunk 4]
```diff
@@ -16,106 +16,56 @@ Encapsulates the GTM / Google Analytics functionality
...
-    </p>
-    <p>
-    As this mixin uses authentication token for some of its data it is needed for it to be added on the page level. There are three public methods available to use:
-    - `preparePlatformData` - evaluates and gather data for the `platformData` GA model.
-    - `prepareUserData` - evaluates and gather data for the `userData` GA model. Can take optional `authToken` prop to push data related to authToken. Without the prop userData will contain only userId.
-    - `preparePageData` - evaluates and gather data for the `pageData` GA model. Can take optional object containing `conversationId, requestId, authToken` prop to push data related to these attributes.
-    - `pushAnalytics` - pushes `platformData` and `userData` to the dataLayer.
+    F-Analytics is a 'Nuxt - Plugin' that needs to be registered in the "nuxt.config.js" but as it is contained in a fozzie service you need to create a local plugin to reference the external plugin, which then allows you to pass in options (see the `Options` section).<br><br>
```
Comment (dandel10n) [inline packages/services/f-analytics/README.md]: Would be useful to add anchor tag for "(see the `Options` section)"

File: packages/services/f-analytics/src/services/analytics.service.js [hunk 5]
```diff
@@ -14,7 +18,9 @@ export default class AnalyticService {
     pushPlatformData () {
         const platformData = { ...this.store.state[`${this.namespace}`].platformData };
 
-        mapPlatformData(platformData, this.featureName, this.locale, this.req);
+        mapPlatformData({
+            platformData, featureName: this.featureName, locale: this.locale, req: this.req
+        });
```
Comment (dandel10n) [inline packages/services/f-analytics/src/services/analytics.service.js]:
```suggestion
        mapPlatformData({
            platformData,
            featureName: this.featureName,
            locale: this.locale,
            req: this.req
        });
```

File: packages/services/f-analytics/src/services/analytics.service.js [hunk 6]
```diff
@@ -37,6 +43,25 @@ export default class AnalyticService {
...
+        pageName,
+        conversationId,
+        requestId,
+        httpStatusCode
+    } = {}) {
+        const pageData = { ...this.store.state[`${this.namespace}`].pageData };
+
+        mapPageData({
+            pageData, featureName: this.featureName, pageName, conversationId, requestId, httpStatusCode
+        });
```
Comment (dandel10n) [inline packages/services/f-analytics/src/services/analytics.service.js]:
```suggestion
        mapPageData({
            pageData,
            featureName: this.featureName,
            pageName,
            conversationId,
            requestId,
            httpStatusCode
        });
```

**Review by dandel10n — COMMENTED** [review]
Looks good, I added just few code style comments and some minor readme amendments

**Review by dandel10n — DISMISSED** [review]
Code looks good to me, but the CI build still needs to be fixed :(

File: packages/services/f-analytics/package.json [hunk 7]
```diff
@@ -26,28 +28,24 @@
     "fozzie"
   ],
   "scripts": {
-    "prepublishOnly": "yarn lint && yarn test && yarn build",
-    "build": "vue-cli-service build --target lib --name f-analytics ./src/index.js",
-    "lint": "vue-cli-service lint",
+    "XprepublishOnly": "yarn lint && yarn test && yarn build",
+    "prepublishOnly": "yarn build",
```
Comment (dandel10n) [inline packages/services/f-analytics/package.json]:
```suggestion
    "prepublishOnly": "yarn lint && yarn test && yarn build",
```

**Review by dandel10n — APPROVED** [review]

**Review by AndyMayJE — APPROVED** [review]

**Pull request merged**